streams/src/test/java/org/apache/kafka/streams/integration/SuppressionIntegrationTest.java

```diff
@@ -0,0 +1,314 @@
+                    .withLoggingDisabled()
+            )
+            .groupBy((k, v) -> new KeyValue<>(v, k), Serialized.with(STRING_SERDE, STRING_SERDE))
+            .count(Materialized.<String, Long, KeyValueStore<Bytes, byte[]>>as("counts").withCachingDisabled());
+
+        valueCounts
+            .suppress(intermediateEvents(withEmitAfter(Duration.ZERO)))
+            .toStream()
+            .to(outputSuppressed, Produced.with(STRING_SERDE, Serdes.Long()));
+
+        valueCounts
+            .toStream()
+            .to(outputRaw, Produced.with(STRING_SERDE, Serdes.Long()));
+
+        final KafkaStreams driver = getCleanStartedStreams(appId, builder);
+
+        try {
+            produceSynchronously(
+                input,
+                asList(
+                    new KVT<>("k1", "v1", scaledTime(0L)),
+                    new KVT<>("k1", "v2", scaledTime(1L)),
+                    new KVT<>("k2", "v1", scaledTime(2L)),
+                    new KVT<>("x", "x", scaledTime(4L))
+                )
+            );
+            verifyOutput(
+                outputRaw,
+                asList(
+                    new KVT<>("v1", 1L, scaledTime(0L)),
+                    new KVT<>("v1", 0L, scaledTime(1L)),
+                    new KVT<>("v2", 1L, scaledTime(1L)),
+                    new KVT<>("v1", 1L, scaledTime(2L)),
+                    new KVT<>("x", 1L, scaledTime(4L))
+                )
+            );
+            verifyOutput(
+                outputSuppressed,
+                asList(
+                    new KVT<>("v1", 1L, scaledTime(0L)),
+                    new KVT<>("v1", 0L, scaledTime(1L)),
+                    new KVT<>("v2", 1L, scaledTime(1L)),
+                    new KVT<>("v1", 1L, scaledTime(2L)),
+                    new KVT<>("x", 1L, scaledTime(4L))
+                )
+            );
+        } finally {
+            driver.close();
+            cleanStateAfterTest(driver);
+        }
+    }
+
+    private void cleanStateBeforeTest(final String... topic) throws InterruptedException {
+        CLUSTER.deleteAllTopicsAndWait(30_000L);
+        for (final String s : topic) {
+            CLUSTER.createTopic(s, 1, 1);
+        }
+    }
+
+    private KafkaStreams getCleanStartedStreams(final String appId, final StreamsBuilder builder) {
+        final Properties streamsConfig = mkProperties(mkMap(
+            mkEntry(StreamsConfig.APPLICATION_ID_CONFIG, appId),
+            mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers()),
+            mkEntry(StreamsConfig.POLL_MS_CONFIG, Objects.toString(COMMIT_INTERVAL)),
+            mkEntry(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, Objects.toString(COMMIT_INTERVAL))
+        ));
+        final KafkaStreams driver = new KafkaStreams(builder.build(), streamsConfig);
+        driver.cleanUp();
+        driver.start();
+        return driver;
+    }
+
+    private void cleanStateAfterTest(final KafkaStreams driver) throws InterruptedException {
```

Comment:
nit: do you need `cleanStateAfterTest`?   In  `getCleanStartedStreams` you already call `driver.cleanUp` and delete all topics in `cleanStateBeforeTest`

Reply:
It's a belt-and-suspenders thing. We want it at the end so that the tests don't leave garbage around (for example the last test). I also added it at the beginning just in case a prior run got forcibly killed and never had a chance to clean up.

Reply:
If you don't trust the regular cleanup, the question is why? Also, should we apply this patter to other tests, too?

Reply:
Specifically, the "if" I described above happens to me when I am debugging integration tests. If you set a breakpoint and then hit the "stop" button, it doesn't get a chance to run the normal "after" cleanup.

Depending on whether the test uses a randomized state directory or not, this will either give you dirty state for the next run, or it'll just consume more and more disk space in `/tmp` until you run out. I have had both happen to me in the last few months.

I do think we should use the same pattern for all the integration tests.